packages/workbox-core/_private/DBWrapper.mjs

```diff
@@ -0,0 +1,351 @@
+    const result = await new Promise((resolve, reject) => {
+      const txn = this._db.transaction(storeNames, type);
+      const done = (value) => resolve(value);
+      const abort = () => {
+        reject(new Error('The transaction was manually aborted'));
+        txn.abort();
+      };
+      txn.onerror = (evt) => reject(evt.target.error);
+      txn.onabort = (evt) => reject(evt.target.error);
+      txn.oncomplete = () => resolve();
+
+      const stores = {};
+      for (const storeName of storeNames) {
+        stores[storeName] = txn.objectStore(storeName);
+      }
+      callback(stores, done, abort);
+    });
+    return result;
+  }
+
+  /**
+   * Delegates async to a native IDBObjectStore method.
+   *
+   * @param {string} method The method name.
+   * @param {string} storeName The object store name.
+   * @param {string} type Can be `readonly` or `readwrite`.
+   * @param {...*} args The list of args to pass to the native method.
+   * @return {*} The result of the transaction.
+   */
+  async _call(method, storeName, type, ...args) {
+    await this.open();
+    const callback = (stores, done) => {
+      stores[storeName][method](...args).onsuccess = (evt) => {
+        done(evt.target.result);
+      };
+    };
+
+    return await this.transaction([storeName], type, callback);
+  }
+
+  /**
+   * The default onversionchange handler, which closes the database so other
+   * connections can open without being blocked.
+   *
+   * @param {Event} evt
+   */
+  _onversionchange(evt) {
+    this.close();
+  }
+
+  /**
+   * Closes the connection opened by `DBWrapper.open()`. Generally this method
+   * doesn't need to be called since:
+   *   1. It's usually better to keep a connection open since opening
+   *      a new connection is somewhat slow.
+   *   2. Connections are automatically closed when the reference is
+   *      garbage collected.
+   * The primary use case for needing to close a connection is when another
+   * reference (typically in another tab) needs to upgrade it and would be
+   * blocked by the current, open connection.
+   */
+  close() {
+    if (this._db) this._db.close();
+  }
+
+  /**
+   * Deletes the database.
+   */
+  async deleteDatabase() {
+    this.close();
+    // TODO(philipwalton): add a timeout to handle the blocked case.
+    await new Promise((resolve, reject) => {
+      const request = indexedDB.deleteDatabase(this._name);
+      request.onerror = (evt) => reject(evt.target.error);
+      request.onsuccess = (evt) => resolve();
+    });
+  }
+
+  /**
+   * A helper that makes it easier to run IDBRequests in parallel. This
+   * method takes an array of IDBRequests and runs the callback once all of
+   * them are successful. This method should be used along with the
+   * `transaction()` so any errors are automatically handled.
+   *
+   * @param {Array<IDBRequest>} requests
+   * @param {function(Array<*>):undefined} callback
+   */
+  static onsuccessAll(requests, callback) {
```

Comment:
This is one of these cases where I can't use promises at all because in Safari/FF the transaction will complete before the promise resolves.

But as I say in https://github.com/GoogleChrome/workbox/pull/968#discussion_r148584683, since this is just a helper (and not actually used in any of our code) I will remove it.